### PR TITLE
added `VecSimIndex_EstimateInitialSize()` function

### PR DESCRIPTION
**Pull request by GuyAv46**
added this function for estimating the size of a vector index before creating it

### CODE REVIEW

File: tests/unit/test_bruteforce.cpp [hunk 1]
```diff
@@ -1242,3 +1242,22 @@ TEST_F(BruteForceTest, testCosine) {
...
+                                             .metric = VecSimMetric_Cosine,
+                                             .initialCapacity = n}};
+    size_t test_estimation = sizeof(BruteForceIndex);
+
+    test_estimation += n * (sizeof(idType) + sizeof(labelType)); // labelToIdLookup
+    test_estimation += n * sizeof(idType);                       // idToVectorBlockMemberMapping
+
+    size_t estimation = VecSimIndex_EstimateSize(&params);
+    ASSERT_GE(estimation, test_estimation);
+    ASSERT_LE(estimation, test_estimation + 50);
```
Comment (alonre24) [inline tests/unit/test_bruteforce.cpp:1262]: Can you explain what is the reason for that `estimation` is in `[test_estimation, test_estimation+50]` (and not equals to `test_estimation`, and how did you infer it?

File: tests/unit/test_hnswlib.cpp [hunk 2]
```diff
@@ -1353,4 +1353,29 @@ TEST_F(HNSWLibTest, testCosine) {
...
+    // link lists
+    test_estimation += sizeof(void *) * n;
+    // main data block
+    size_t size_links_level0 = sizeof(linklistsizeint) + M * 2 * sizeof(tableint) + sizeof(void *);
+    size_t size_data_per_element = size_links_level0 + dim * sizeof(float) + sizeof(labeltype);
+    test_estimation += n * size_data_per_element;
+
+    size_t estimation = VecSimIndex_EstimateSize(&params);
+    ASSERT_GE(estimation, test_estimation);
+    ASSERT_LE(estimation, test_estimation + 32);
```
Comment (alonre24) [inline tests/unit/test_hnswlib.cpp:1379]: Can you explain what is the reason for that `estimation` is in `[test_estimation, test_estimation+32]` (and not equals to `test_estimation`, and how did you infer it?

File: src/VecSim/vec_sim.h [hunk 3]
```diff
@@ -19,6 +19,14 @@ typedef struct VecSimIndex VecSimIndex;
  */
 VecSimIndex *VecSimIndex_New(const VecSimParams *params);
 
+/**
+ * @brief Estimates the size of an empty index according to the parameters.
+ * @param params index configurations (initial size, data type, dimension, metric, algorithm and the
+ * algorithm-related params).
+ * @return Estimated index size.
+ */
+size_t VecSimIndex_EstimateSize(const VecSimParams *params);
```
Comment (alonre24) [inline src/VecSim/vec_sim.h:28]: Consider renaming to `VecSimIndex_EstimateInitialSize`

**Review by alonre24 — COMMENTED** [review]
Nice!

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**